aws/table_aws_glue_catalog_table.go

```diff
@@ -0,0 +1,255 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+
+	"github.com/turbot/steampipe-plugin-sdk/v3/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsGlueCatalogTable(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_glue_catalog_table",
+		Description: "AWS Glue Catalog Table",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.AllColumns([]string{"name", "database_name"}),
+			ShouldIgnoreError: isNotFoundError([]string{"EntityNotFoundException"}),
+			Hydrate:           getGlueCatalogTable,
+		},
+		List: &plugin.ListConfig{
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "catalog_id", Require: plugin.Optional},
+				{Name: "database_name", Require: plugin.Optional},
+			},
+			ParentHydrate: listGlueCatalogDatabases,
+			Hydrate:       listGlueCatalogTables,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The table name.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "catalog_id",
+				Description: "The ID of the Data Catalog in which the table resides.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "create_time",
+				Description: "The time when the table definition was created in the data catalog.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "description",
+				Description: "A description of the table.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "created_by",
+				Description: "The person or entity who created the table.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "database_name",
+				Description: "The name of the database where the table metadata resides.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "is_registered_with_lake_formation",
+				Description: "Indicates whether the table has been registered with lake formation.",
+				Type:        proto.ColumnType_BOOL,
+			},
+			{
+				Name:        "last_access_time",
+				Description: "The last time that the table was accessed. This is usually taken from HDFS, and might not be reliable.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "last_analyzed_time",
+				Description: "The last time that column statistics were computed for this table.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "owner",
+				Description: "The owner of the table.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "retention",
+				Description: "The retention time for this table.",
+				Type:        proto.ColumnType_STRING,
```

Comment:
@karanpopat Why is this a string? The API shows this field as a number instead. If it's a number, the example would also not need casting.